pkg/reconciler/tenancy/initialization/apibinder/bootstrap.go

```diff
@@ -0,0 +1,276 @@
+	}
+	klog.Infof("APIBinding %q|%q is bound", installInto.String(), binding.Name)
+
+	cwt, err := clusterWorkspaceType()
+	if err != nil {
+		return fmt.Errorf("could not resolve ClusterWorkspaceType: %w", err)
+	}
+
+	tenancyClient := kcpClient.TenancyV1alpha1()
+	cwt, err = tenancyClient.ClusterWorkspaceTypes().Create(ctx, cwt, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("could not create ClusterWorkspaceType: %w", err)
+	}
+	klog.Infof("Created ClusterWorkspaceType %q|%q", installInto.String(), cwt.Name)
+
+	if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: 10 * time.Millisecond,
+		Factor:   2,
+		Steps:    10,
+	}, func() (done bool, err error) {
+		cwt, err = tenancyClient.ClusterWorkspaceTypes().Get(ctx, cwt.Name, metav1.GetOptions{})
+		if err != nil {
+			klog.Infof("Error waiting for ClusterWorkspaceType %q|%q virtual workspace URLs to be ready: %v", installInto.String(), cwt.Name, err)
+			return false, err
+		}
+		return checkConditionTrueWhileLogging(cwt, tenancyv1alpha1.ClusterWorkspaceTypeVirtualWorkspaceURLsReady, fmt.Sprintf("Not done waiting for ClusterWorkspaceType %q|%q virtual workspace URLs to be ready", installInto.String(), cwt.Name)), nil
+	}); err != nil {
+		return fmt.Errorf("could not wait for virtual workspace URLs to be ready on ClusterWorkspaceType: %w", err)
+	}
+	klog.Infof("ClusterWorkspaceType %q|%q virtual workspace URLs are ready", installInto.String(), cwt.Name)
+
+	initializerControllerCfg := rest.AddUserAgent(rest.CopyConfig(cfg), "kcp-apibinder-initializer-bootstrapper-controller")
+
+	initializingWorkspacesCfg := rest.CopyConfig(initializerControllerCfg)
+	initializingWorkspacesCfg.Host = cwt.Status.VirtualWorkspaces[0].URL
+	initializingWorkspacesKcpClusterClient, err := kcpclient.NewClusterForConfig(initializingWorkspacesCfg)
+	if err != nil {
+		return err
+	}
+	initializingWorkspacesKcpInformers := kcpinformer.NewSharedInformerFactory(initializingWorkspacesKcpClusterClient.Cluster(logicalcluster.Wildcard), 10*time.Minute)
+
+	apiExportCfg := rest.CopyConfig(initializerControllerCfg)
+	apiExportCfg.Host = export.Status.VirtualWorkspaces[0].URL
+	apiExportInitializationClusterClient, err := initializationclient.NewClusterForConfig(apiExportCfg)
+	if err != nil {
+		return err
+	}
+	apiExportInitializationInformers := initializationinformer.NewSharedInformerFactory(apiExportInitializationClusterClient.Cluster(logicalcluster.Wildcard), 10*time.Minute)
+
+	initializerController, err := NewAPIBinder(
+		initializationutil.InitializerForType(cwt),
+		initializingWorkspacesKcpClusterClient,
+		initializingWorkspacesKcpInformers.Tenancy().V1alpha1().ClusterWorkspaces(),
+		apiExportInitializationInformers.Initialization().V1alpha1().APISets(),
+	)
+	if err != nil {
+		return err
+	}
+
+	kcpInformers := kcpinformer.NewSharedInformerFactory(kcpClusterClient.Cluster(logicalcluster.Wildcard), 10*time.Minute)
```

Comment:
Does this need to be different from the shared factory we already have in the server?

Reply:
No, I just wanted it to be standalone / simple as possible for the first pass.